### PR TITLE
refactor show_traces in memory_tracker

### PR DESCRIPTION
**Pull request by zhaojuanmao**
refactor show_tracers in memory_tracker to make it plot multiple figures and also can load serialized stats and then plot figures

### CODE REVIEW

File: torch/distributed/_tools/memory_tracker.py [hunk 1]
```diff
@@ -147,36 +143,86 @@ def summary(self, top: int = 20) -> None:
         print("------------------------------------------------")
 
     @no_type_check
-    def show_traces(self) -> None:
+    def show_traces(self, path: str = "") -> None:
```
Comment (awgu) [inline torch/distributed/_tools/memory_tracker.py:146]: Sorry, maybe these PRs are not ready for review. This diff to `show_traces()` looks different than the one in https://github.com/pytorch/pytorch/pull/90144.

File: torch/distributed/_tools/memory_tracker.py [hunk 2]
```diff
@@ -145,36 +145,55 @@ def summary(self, top: int = 20) -> None:
...
 
     @no_type_check
-    def show_traces(self) -> None:
-        """
-        Show the traces of ``memory_allocated``, ``memory_active`` and ``memory_reserved`` at
-        operator level and the marker 'fw_bw_boundary' at the boundary of forward pass
-        and backward pass.
-        """
+    def show_traces(self, path: str = "") -> None:
+        from itertools import chain
```
Comment (rohan-varma) [inline torch/distributed/_tools/memory_tracker.py:149]: nit: import at the top?

**Review by rohan-varma — APPROVED** [review]
LGTM

**Pull request closed without merging**